### PR TITLE
TrustX Bid Adapter: Use new format as default + new format for keywords

### PR DESCRIPTION
**Pull request by PWyrembak**
## Type of change

- [x] Feature

## Description of change

Use new format as default + new format for keywords

- paul@trustx.org
- [x] official adapter submission

### CODE REVIEW

**Comment by TheMediaGrid** [comment]
hey team, any chance we could get this one reviewed? @msm0504 
we really need this one to be deployed this week

**Comment by ChrisHuie** [comment]
@TheMediaGrid this is a pretty large pr and rewrite so could take a bit to go through.

File: modules/trustxBidAdapter.js [hunk 1]
```diff
@@ -313,6 +449,51 @@ function addSegments(name, segName, segments, data, bidConfigName) {
   }
 }
 
+function reformatKeywords(pageKeywords) {
```
Comment (Fawke) [inline modules/trustxBidAdapter.js:452]: I see that a bunch of validations is done on the ortb2/fpd object here. Just wanted to let you know that a prebid.js optional modlue [validationFpdModule](https://docs.prebid.org/dev-docs/modules/validationFpdModule.html) was released recently. You may still need this if it's not included in the build.

Do you need to reformat them?

**Review by Fawke — CHANGES_REQUESTED** [review]
Hi @piwanczak,

Changes mostly look good. For testing, I wanted to receive a test bid response from your endpoint, but none of the examples currently given in the md file works. Can you please update them (both video and banner). Also, if you can update the examples to include the required `mediaTypes` property, that'll be great.

Apart from that, please check the inline comment.

**Comment by PWyrembak** [comment]
Hello! We have fixed the md file. As for the validation of keywords, you probably misunderstood, we validate mostly keywords from the adUnit parameters

**Review by Fawke — APPROVED** [review]
LGTM

**Pull request merged**